iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl

```diff
@@ -503,6 +503,36 @@ inline
     return *this;
 }
 
+template <uint64_t Capacity>
+template <typename T>
+inline constexpr typename std::enable_if<internal::IsCharArray<T>::value || internal::IsCxxString<T>::value, bool>::type
+string<Capacity>::insert(const uint64_t pos, const T& str, const uint64_t count) noexcept
+{
+    if (count > internal::GetSize<T>::call(str))
+    {
+        return false;
+    }
+    if (pos > m_rawstringSize)
+    {
+        return false;
+    }
+    const auto new_size = m_rawstringSize + count;
```

Comment:
I think you can get rid of the two previous early returns with this return. This should cover everything.

Reply:
Unfortunately no. `count` could still be greater than string to be inserted. This is allowed in `std::string` but I decided against because it's undefined what will be inserted. `pos` could be greater than `m_rawStringSize` too. Since this is tested in `substr`, I moved the return after creating the `remainder` substring.